engine/src/main/scala/org/scalarules/dsl/core/glossaries/Glossary.scala

```diff
@@ -11,7 +11,7 @@ import scala.language.experimental.macros
   * Utility base class for collecting and namespacing `Fact`s. You can extend this class, define facts in it and receive a
   * utility collection of all facts declared in your class.
   */
-class MacroGlossary {
+class Glossary {
   def defineFact[A](): SingularFact[A] = macro FactMacros.defineFactMacroImpl[A]
   def defineFact[A](description: String = "No description"): SingularFact[A] = macro FactMacros.defineFactMacroWithDescriptionImpl[A]
```

Comment:
In "normal" Scala-world, you would be completely correct. However, macros do not support default parameters. So, in order to support the `description` being optional, we need both these macros, one with a `description` and one without a `description` parameter.

Macro declarations need to match 1-on-1 with the signature of their macro implementation.